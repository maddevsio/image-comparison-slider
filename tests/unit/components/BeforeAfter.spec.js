import BeforeAfter from '@/components/BeforeAfter'
import { shallowMount } from '@vue/test-utils'

const props = {
  afterImage: 'img.jpg',
  beforeImage: 'img.jpg',
  value: 50,
  step: 0.1,
}

const spy = jest.fn()
const testWidth = 420
const inputValue = 89

describe('BeforeAfter component', () => {
  let wrapper

  beforeAll(() => {
    window.addEventListener('resize', spy)
  })

  beforeEach(() => {
    wrapper = shallowMount(BeforeAfter, {
      propsData: props,
    })
  })

  afterEach(() => {
    wrapper = null
  })

  it('should render correctly with images', () => {
    expect(wrapper.props()).toEqual(props)
    expect(wrapper.is(BeforeAfter)).toBe(true)
    expect(wrapper).toMatchSnapshot()
  })

  it('the compareWidth should correctly react to change value of the input', () => {
    const rangeInput = wrapper.find('.compare__range')
    rangeInput.setValue(inputValue)
    rangeInput.trigger('input change')

    setTimeout(() => {
      expect(wrapper.find('.img-wrapper__compare-overlay').attributes().style).toBe(`width: ${inputValue}%;`)
    }, 0)
    expect(wrapper.vm.compareWidth).toBe(`${inputValue}`)
  })

  it('does not fire resize event by default', () => {
    expect(spy).not.toHaveBeenCalled()
    expect(window.innerWidth).not.toBe(testWidth)
  })

  it('updates the window width', () => {
    window.innerWidth = testWidth
    window.dispatchEvent(new Event('resize'))

    expect(spy).toHaveBeenCalledTimes(1)
    expect(window.innerWidth).toBe(testWidth)
  })

  it('should correctly remove resize event listener from window', () => {
    wrapper = shallowMount(BeforeAfter, {
      propsData: props,
      attachTo: document.body,
    })
    wrapper.destroy()

    expect(spy).toHaveBeenCalledTimes(1)
  })
})
