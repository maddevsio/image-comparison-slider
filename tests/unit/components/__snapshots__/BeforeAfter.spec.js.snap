// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeforeAfter component should render correctly with images 1`] = `
<div class="images-wrapper"><img src="img.jpg" alt="Before image" class="before-image">
  <div class="compare-overlay" style="width: 50%;"><img src="img.jpg" alt="After image" class="after-image" style="width: 0px;"></div> <input type="range" min="0" max="100" step="0.1" tabindex="-1" class="compare__range">
  <div class="handle-wrap" style="left: 50%;">
    <div class="handle"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="handle__arrow handle__arrow--l">
        <polyline points="15 18 9 12 15 6"></polyline>
      </svg> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="handle__arrow handle__arrow--r">
        <polyline points="9 18 15 12 9 6"></polyline>
      </svg></div> <span class="handle-line"></span>
  </div>
</div>
`;
